src/vivarium_conic_sqlns/components/observers.py

```diff
@@ -9,3 +13,91 @@ def setup(self, builder):
+                'sample_size': 100,
+                'path': '/dev/null'
+            }
+        }
+    }
+
+    @property
+    def name(self):
+        return "sample_history_observer"
+
+    def __init__(self):
+        self.history_snapshots = []
+        self.sample_index = None
+
+    def setup(self, builder):
+        config = builder.configuration['metrics']['sample_history']
+        self.sample_size = config['sample_size']
+        self.path = pathlib.Path(config['path']).resolve()
+        if self.path.suffix != '.hdf':
+            raise ValueError("metrics: sample_history: path must specify a path to an HDF file.")
+
+        self.clock = builder.time.clock()
+        self.randomness = builder.randomness.get_stream('sample_index')
+
+        self.builder = builder
+        self.population_view = builder.population.get_view(['alive', 'age', 'sex', 'exit_time'])
+
+        self.pipelines = {
+            'iron_deficiency_exposure': builder.value.get_value('iron_deficiency.exposure'),
+            'child_stunting_exposure': builder.value.get_value('child_stunting.exposure'),
+            'child_wasting_exposure': builder.value.get_value('child_wasting.exposure'),
+            'lower_resipratory_infections_incidence_rate': builder.value.get_value('lower_respiratory_infections.incidence_rate'),
+            'diarrheal_diseases_incidence_rate' : builder.value.get_value('diarrheal_diseases.incidence_rate'),
+            'measles_incidence_rate': builder.value.get_value('measles.incidence_rate'),
+            'disability_weight': builder.value.get_value('disability_weight')
+        }
+
+        builder.population.initializes_simulants(self.get_sample_index)
+        builder.event.register_listener('collect_metrics', self.record)
+        builder.event.register_listener('simulation_end', self.dump_history)
+
+    def get_sample_index(self, pop_data):
+        if self.sample_index is None:
+            pop_size = len(pop_data.index)
+            if self.sample_size > pop_size:
+                raise ValueError("Sample size cannot exceed initial population size.")
+
+            draw = self.randomness.get_draw(pop_data.index)
+            priority_index = [i for d, i in sorted(zip(draw, pop_data.index), key=lambda x:x[0])]
+            self.sample_index = pd.Index(priority_index[:self.sample_size])
+
+    def record(self, event):
+        pop = self.population_view.get(self.sample_index)
+        pipeline_results = []
+        for name, pipeline in self.pipelines.items():
+            if name == 'iron_deficiency':
+                skip_post_processor = True  # raw hemoglobin was requested
+            else:
+                skip_post_processor = False
+
+            # Using pop.index due to untracked individuals
+            values = pipeline(pop.index, skip_post_processor=skip_post_processor)
+            values = values.rename(name)
+            pipeline_results.append(values)
+
+            raw_values = pipeline.source(pop.index)
```

Comment:
If name != disability weight

Reply:
No disability weight pipeline sources, they're super confusing.

Reply:
No incidence pipelines (Both SOURCES) either, pipeline.source is risk-deleted, not un-intervened.